keras_nlp/samplers/top_k_sampler_test.py

```diff
@@ -94,6 +95,21 @@ def next(prompt, state, index):
         output_ids = set(output[0].numpy())
         self.assertContainsSubset(output_ids, range(5))
 
+    def test_outputs_in_top_k_with_temperature(self):
+        def next(prompt, state, index):
+            # Return a distribution where each id is progressively less likely.
+            logits = tf.range(self.vocab_size, 0, -1, dtype="float32")
+            logits = tf.repeat(logits[tf.newaxis, :], self.batch_size, axis=0)
+            return logits, state
+
+        prompt = tf.fill((self.batch_size, self.length), self.char_lookup["z"])
+        output = self.sampler_temperature(
+            next=next,
+            prompt=prompt,
+        )
+        output_ids = set(output[0].numpy())
```

Comment:
no need to do set here, just `self.assertAllEqual(output, [0] * self.length)` or something like that.